### ISSUE TITLE
feature: Encrypt user OpenAPI keys

### ISSUE THREAD

**Issue by cerealkill** [windmill-labs/windmill#3847]
Hey dev team,

It would be really nice and useful to secure the user OpenAPI credentials, and others that I may not have noticed yet, like a regular secret variable. When syncing my files with `wmill` I have noticed that they are not encrypted like the secrets are, and that poses a financial risk. Thank you.

Before I go, props: Thanks for open sourcing windmill, I really love it, and may support the team with a couple Enterprise accounts in the future, once business picks up.

**Comment by rubenfiszel** [comment]
The openai key is encrypted by default:

![image](https://github.com/windmill-labs/windmill/assets/275584/d07af81c-6ded-4312-b85c-94695be1f659)

You can see the Secret toggle there, which mean on creation it will store it as a separate secret variable

**Comment by rubenfiszel** [comment]
Actually if you add the key from the workspace creation modal, we may have had the oversight to not encrypt it by default, we will update with our findings here. Thanks for the report

**Comment by cerealkill** [comment]
Hey Ruben,

Thanks for the quick reply. It is actually quite reassuring to receive an answer directly from you. I appreciate you taking the time.

When I installed windmill I was using some few versions back than latest, and I added the credentials right when I created the user. Maybe it is still the case, but when I used `wmill sync pull` to backup my workspace, my credentials came back in plain text. Here:

<img width="515" alt="Screenshot 2024-05-29 at 19 45 08" src="https://github.com/windmill-labs/windmill/assets/2923413/48df188c-2d06-4c7e-a3a2-5eddf6cde47d">

To fix it, I actually had to create two secret variables, and point them to the OpenApi resource.

<img width="787" alt="Screenshot 2024-05-29 at 19 48 17" src="https://github.com/windmill-labs/windmill/assets/2923413/d3cd0658-6fe1-4444-917f-9b78e254dff1">

And now when I sync there are no plain text variables no more. It would be nice to be the default behaviour, and maybe it is in the newer versions already.

Another feature request work mentioning and related to the variables, is to subscript a json, so that I could have a secret variable for my openapi credentials as one json. i.e.: `$var:f/openai/api-credentials{key}` and `$var:f/openai/api-credentials{org}`. I understand that is simply convenience, and there are bigger fish to fry.

All the best to you and Windmill as an org. 

Cheers,
Paul
